### ISSUE TITLE
The task is not stopped when "await BackgroundJob.stop();"

### ISSUE THREAD

**Issue by gabriel1lima** [Rapsssito/react-native-background-actions#11]
Using the sample application, I added a new TouchableOpacity to call the function "BackgroundJob.stop ()", but when I click on it the loop still sends console.log with "Runned -> index"

**Comment by Rapsssito** [comment]
@gabriel1lima, you are right, thanks for the feedback! I will release a fix as soon as possible.

**Comment by gabriel1lima** [comment]
Thanks!
Why does it happen? Could you give me a quick solution where I could mess with the lib code and make it work, because I am in urgent need of this functionality that your lib offers

**Comment by Rapsssito** [comment]
@gabriel1lima, it happens because when the task is "killed", but the promise has not finished, **the "killing" process will wait until the promise resolves** (_maybe indefinitely_). The React Native [Headless JS docs](https://facebook.github.io/react-native/docs/headless-js-android#the-js-api) do not say much about it.

A solution could be to subscribe the background action to an event that causes the promise to resolve. Example code:

```javascript
const eventEmitter = new EventEmitter();

const backgroundTask = async (asyncData) => {
  await new Promise((resolve) => {
     eventEmitter.addListener('close', () => resolve());
     // Do other stuff
  };
}

// Anywhere else
eventEmitter.emit('close');
```

**Comment by gabriel1lima** [comment]
I didn't understand where I should add this code, you could exemplify.

**Comment by Rapsssito** [comment]
**I have updated the [example code](https://github.com/Rapsssito/react-native-background-actions/blob/master/examples/backgroundExample/App.js)** with a more functional one, it may serve you as an example. However, it does not work perfectly.

I am working on an improved API and more useful (non-serializable) parameter handling. Keep in mind that the example might be updated in a few hours.

**Comment by gabriel1lima** [comment]
Thanks!

**Comment by Rapsssito** [comment]
@gabriel1lima, I have released a **new version ([v2.1.0](https://github.com/Rapsssito/react-native-background-actions/releases/tag/v2.1.0))**, updated the **usage [documentation](https://github.com/Rapsssito/react-native-background-actions#usage)** and refactor the example code removing the need for an `EventEmitter`.

I hope this fixes your issue, feel free to reopen this issue if your problem persists.

**Comment by Phasor1** [comment]
Hi @Rapsssito, i confirm that the BackgroundJob.stop() works, but i have a similar problem to this closed issue: putting BackgroundJob.stop() inside a componentWillUnmount when the app close, the promise doesn't end, i suppose beacouse the app close before the promise resolves. There's a way be shure that the app doesn't close before the promise ends, or make the promise ends by another way?
